crates/bevy_render/src/render_event.rs

```diff
@@ -0,0 +1,121 @@
+use core::marker::PhantomData;
+
+use async_channel::{Receiver, Sender};
+use bevy_app::{App, Plugin, PreUpdate};
+use bevy_ecs::{
+    change_detection::MaybeLocation,
+    event::{Event, Events},
+    resource::Resource,
+    system::{Res, ResMut, SystemParam},
+};
+
+use crate::RenderApp;
+
+pub trait RenderEventApp {
```

Comment:
Not sure what our convention is here -- should this be in the prelude so it doesn't have to be imported?